src/ResourceManager/Automation/Commands.Automation.Test/ScenarioTests/SourceControlTests.ps1

```diff
@@ -0,0 +1,322 @@
+
+        # Validate that the source control was created
+        Assert-NotNull $createdSourceControl "Failed to create VsoGit source control."
+
+        # Validate the source control properties
+        $propertiesToValidate = @("Name", "RepoUrl", "SourceType", "FolderPath")
+        
+        foreach ($property in $propertiesToValidate)
+        {
+            Assert-AreEqual $sourceControl.$property $createdSourceControl.$property `
+                "'$property' of created source control does not match. Expected:$($sourceControl.$property) Actual: $($createdSourceControl.$property)"
+        }
+
+        # Enable PublishRunbook in the newly created source control 
+        $updatedSourceControl = Update-AzureRmAutomationSourceControl -ResourceGroupName $resourceGroupName `
+                                                                      -AutomationAccountName $automationAccountName  `
+                                                                      -Name $sourceControl.Name  `
+                                                                      -PublishRunbook $true
+        $expectedPropertyValue = "True"
+        Assert-AreEqual $updatedSourceControl.PublishRunbook $expectedPropertyValue `
+                "'PublishRunbook' property does not match. Expected: $expectedPropertyValue. Actual: $($updatedSourceControl.PublishRunbook)"
+        
+        # Start a sync for the source control
+        $jobId = "27dcdb17-1f65-42e9-9eeb-088a5f50eeb8"
+        Start-AzureRmAutomationSourceControlSyncJob -ResourceGroupName $resourceGroupName `
+                                            -AutomationAccountName $automationAccountName  `
+                                            -Name $sourceControl.Name `
+                                            -JobId $jobId
+
+        WaitForSourceControlSyncJobState -Name $sourceControl.Name -JobId $jobId -ExpectedState Completed
+
+        # Get the SourceControlSyncJob streams
+        $streams =  Get-AzureRmAutomationSourceControlSyncJobOutput -ResourceGroupName $resourceGroupName `
+                                                                    -AutomationAccountName $automationAccountName  `
+                                                                    -Name $sourceControl.Name `
+                                                                    -JobId $jobId `
+                                                                    -Stream Output | % Summary
+        
+        Assert-True {$streams.count -gt 0} "Failed to get Output stream via Get-AzureRmAutomationSourceControlSyncJobOutput "
+    }
+    finally
+    {
+        EnsureSourceControlDoesNotExist -Name $sourceControl.Name
+    }
+}
+
+function Test-CreateGitHubSourceControlAndSync
+{
+    # GitHub repo info
+    $sourceControl = @{
+        Name = "AASourceControl-GitHub"
+        RepoUrl = "https://github.com/Francisco-Gamino/SwaggerAndCmdletsTests.git"
```

Comment:
see above - assume that the comments aboud hard coded ids, repos, etc. apply throughout the tests